Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -0,0 +1,462 @@
+#if canImport(OSLog)
+import OSLog
+#endif
+#if !os(macOS)
+import UIKit
+#else
+import AppKit
+#endif
+import Combine
+import Foundation
+
+/// Represent  different data sources available for synchronization.
+public enum DataSource {
+	/// Using `key.suite` as data source.
+	case local
+	/// Using `NSUbiquitousKeyValueStore` as data source.
+	case remote
+}
+
+private enum SyncStatus {
+	case start
+	case isSyncing
+	case finish
+}
```

Comment:
This would be more idiomatic Swift:

```suggestion
private enum SyncStatus {
    case idle
    case syncing
    case completed
}
```

Not sure `idle` is correct though.